packages/core/src/button/Button.tsx

```diff
@@ -66,6 +67,20 @@ export interface ButtonProps extends ComponentPropsWithoutRef<"button"> {
    * @since 1.36.0.
    */
   sentiment?: ButtonSentiment;
+
+  /**
+   * If `true`, the button will be in a loading state. This allows a spinner to be nested inside the button.
+   *
+   * @since 1.38.0.
+   */
+  loading?: boolean;
+
+  /**
+   * Text to be announced by screen readers, intended to be used in conjunction with the `loading` prop.
+   *
+   * @since 1.38.0.
+   */
+  announcement?: string;
```

Comment:
We should consider calling this loadingAnnouncement instead if it's supposed to be used with loading

Reply:
Changed it to `loadingAnnouncement`